trackintel/preprocessing/staypoints.py

```diff
@@ -152,35 +146,22 @@ def generate_locations(
         # filter staypoints not belonging to locations
         locs = locs.loc[locs["location_id"] != -1]
 
-        locs["center"] = None  # initialize
-        # locations with only one staypoints is of type "Point"
-        point_idx = locs.geom_type == "Point"
-        if not locs.loc[point_idx].empty:
-            locs.loc[point_idx, "center"] = locs.loc[point_idx, locs.geometry.name]
-        # locations with multiple staypoints is of type "MultiPoint"
-        if not locs.loc[~point_idx].empty:
-            locs.loc[~point_idx, "center"] = locs.loc[~point_idx, locs.geometry.name].apply(
-                lambda p: Point(np.array(p)[:, 0].mean(), np.array(p)[:, 1].mean())
-            )
+        # TODO: mean of degrees is not centroid of area (good enough for our purpose?)
+        locs["center"] = locs.geometry.centroid
 
         # extent is the convex hull of the geometry
-        locs["extent"] = None  # initialize
-        if not locs.empty:
-            locs["extent"] = locs[locs.geometry.name].apply(lambda p: p.convex_hull)
-            # convex_hull of one point would be a Point and two points a Linestring,
-            # we change them into Polygon by creating a buffer of epsilon around them.
-            pointLine_idx = (locs["extent"].geom_type == "LineString") | (locs["extent"].geom_type == "Point")
-
-            if not locs.loc[pointLine_idx].empty:
-                # Perform meter to decimal conversion if the distance metric is haversine
-                if distance_metric == "haversine":
-                    locs.loc[pointLine_idx, "extent"] = locs.loc[pointLine_idx].apply(
-                        lambda p: p["extent"].buffer(meters_to_decimal_degrees(epsilon, p["center"].y)), axis=1
-                    )
-                else:
-                    locs.loc[pointLine_idx, "extent"] = locs.loc[pointLine_idx].apply(
-                        lambda p: p["extent"].buffer(epsilon), axis=1
-                    )
+        locs["extent"] = locs.geometry.convex_hull
+        # convex_hull of Point is Point, and MultiPoint with two Points is a LineString
+        # we change them into Polygon by creating a buffer of epsilon around them.
+        pointLine_idx = (locs["extent"].geom_type == "LineString") | (locs["extent"].geom_type == "Point")
+
+        # Perform meter to decimal conversion if the distance metric is haversine
+        if distance_metric == "haversine":
+            locs.loc[pointLine_idx, "extent"] = locs.loc[pointLine_idx].apply(
+                lambda p: p["extent"].buffer(meters_to_decimal_degrees(epsilon, p["center"].y)), axis=1
```

Comment:
I think this line causes problems when pointLine_idx only contains false. I tried something similar as an example:
```
>> test = pd.DataFrame(np.random.rand(10,2))
>> test.loc[np.zeros(10).astype(bool), 0] = test.loc[np.zeros(10).astype(bool)].apply(lambda x: x[0]*2, axis=1) 
ValueError: Incompatible indexer with DataFrame
```
So I think the same would happen here, maybe you can test it with a set of staypoints that does not lead to LineStrings or Points? Maybe by running DBSCAN with `min_samples=3` and a high epsilon.

Reply:
That is a good catch. But I think it works in this case but I am not completely sure why :D My guess would be something with `0` as index.
This is my test code:
```python
import geopandas as gpd
import pandas as pd
gdf = gpd.GeoDataFrame.from_file(gpd.datasets.get_path("naturalearth_lowres"))
f = pd.Series(False, index=gdf.index)
gdf["extent"] = gdf.convex_hull
gdf.loc[f, "extent"] = gdf.loc[f].apply(
	lambda p: p["extent"].buffer(meters_to_decimal_degrees(epsilon, p["center"].y)),
	axis=1
)
```

Reply:
Okay funny :D Well, maybe you can double check by simply changing one of the unit tests of this function to use `num_samples=3`, because then there won't be points or lines. But should be fine otherwise.